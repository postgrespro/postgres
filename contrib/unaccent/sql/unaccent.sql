CREATE EXTENSION unaccent;

-- must have a UTF8 database
SELECT getdatabaseencoding();
SET client_encoding TO 'KOI8';

SELECT unaccent('foobar');
SELECT unaccent('£ΜΛΑ');
SELECT unaccent('³φιλ');

SELECT unaccent('unaccent', 'foobar');
SELECT unaccent('unaccent', '£ΜΛΑ');
SELECT unaccent('unaccent', '³φιλ');

SELECT ts_lexize('unaccent', 'foobar');
SELECT ts_lexize('unaccent', '£ΜΛΑ');
SELECT ts_lexize('unaccent', '³φιλ');

CREATE TEXT SEARCH CONFIGURATION unaccent(
						COPY=russian
);

ALTER TEXT SEARCH CONFIGURATION unaccent ALTER MAPPING FOR
	asciiword, word WITH unaccent MAP russian_stem;

SELECT to_tsvector('unaccent', 'foobar ³φιλι £ΜΛΙ');
