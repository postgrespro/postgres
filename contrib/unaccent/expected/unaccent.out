CREATE EXTENSION unaccent;
-- must have a UTF8 database
SELECT getdatabaseencoding();
 getdatabaseencoding 
---------------------
 UTF8
(1 row)

SET client_encoding TO 'KOI8';
SELECT unaccent('foobar');
 unaccent 
----------
 foobar
(1 row)

SELECT unaccent('£ΜΛΑ');
 unaccent 
----------
 ΕΜΛΑ
(1 row)

SELECT unaccent('³φιλ');
 unaccent 
----------
 εφιλ
(1 row)

SELECT unaccent('unaccent', 'foobar');
 unaccent 
----------
 foobar
(1 row)

SELECT unaccent('unaccent', '£ΜΛΑ');
 unaccent 
----------
 ΕΜΛΑ
(1 row)

SELECT unaccent('unaccent', '³φιλ');
 unaccent 
----------
 εφιλ
(1 row)

SELECT ts_lexize('unaccent', 'foobar');
 ts_lexize 
-----------
 
(1 row)

SELECT ts_lexize('unaccent', '£ΜΛΑ');
 ts_lexize 
-----------
 {ΕΜΛΑ}
(1 row)

SELECT ts_lexize('unaccent', '³φιλ');
 ts_lexize 
-----------
 {εφιλ}
(1 row)

CREATE TEXT SEARCH CONFIGURATION unaccent(
						COPY=russian
);
ALTER TEXT SEARCH CONFIGURATION unaccent ALTER MAPPING FOR
	asciiword, word WITH unaccent MAP russian_stem;
SELECT to_tsvector('unaccent', 'foobar ³φιλι £ΜΛΙ');
         to_tsvector          
------------------------------
 'foobar':1 'εφιλι':2 'ΕΜΛ':3
(1 row)

